library/Icinga/Web/Navigation/Renderer/RecursiveMenuNavigationRenderer.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/* Icinga Web 2 | (c) 2025 Icinga GmbH | AGPLv3 */
+
+namespace Icinga\Web\Navigation\Renderer;
+
+/**
+ * Renderer for the multi level navigation in the sidebar menu
+ *
+ * @method NavigationRenderer getInnerIterator() {
+ *     {@inheritdoc}
+ * }
+ */
+class RecursiveMenuNavigationRenderer extends RecursiveNavigationRenderer implements NavigationRendererInterface
+{
+    public function beginChildren(): void
+    {
+        parent::beginChildren();
+
+        $currentItemName = $this->getInnerIterator()->current()->getParent()->getName();
+
+        if ($this->getDepth() + 1 > 1) {
```

Comment:
a) Here are children processed
b) The current name is accessed using the parent

Due to both, this check is redundant and always true.